blueprints/structural_sections/concrete/reinforced_concrete_sections/reinforcement_configurations.py

```diff
@@ -0,0 +1,214 @@
+        line : LineString
+            Representing the path of the reinforcement in the section.
+            Start of the line defines the first rebar of the configuration, end of the line defines the last rebar.
+
+        Returns
+        -------
+        List[Rebar]
+            List of Rebar objects.
+        """
+
+
+@dataclass(kw_only=True, frozen=True)
+class ReinforcementByDistance(ReinforcementConfiguration):
+    """Representation of a reinforcement configuration given by center-to-center distance.
+    For example ⌀16-150, ⌀20-200, ⌀25-250, ⌀32-300, etc.
+
+    Parameters
+    ----------
+    center_to_center : MM
+        Maximum center-to-center distance between rebars [mm].
+    """
+
+    center_to_center: MM
+
+    def __post_init__(self) -> None:
+        """Post initialization of the reinforcement configuration."""
+        super().__post_init__()
+        self._validations()
+
+    def _validations(self) -> None:
+        """Validation of the reinforcement configuration."""
+        # center-to-center is at least the diameter
+        if self.center_to_center < self.diameter:
+            msg = f"Center-to-center distance must be at least the diameter of the rebar, got {self.center_to_center}"
+            raise ValueError(msg)
+
+    @property
+    def area(self) -> MM2_M:
+        """Area of the reinforcement configuration per meter [mm²/m]."""
+        return 0.25 * np.pi * self.diameter**2 * (M_TO_MM / self.center_to_center)
+
+    @property
+    def n_rebars_per_meter(self) -> DIMENSIONLESS:
+        """Number of rebars per meter [1/m]."""
+        return 1.0 * M_TO_MM / self.center_to_center
+
+    def to_rebars(self, line: LineString) -> list[Rebar]:
+        """Convert the reinforcement configuration to a list of rebars.
+
+        Parameters
+        ----------
+        line : LineString
+            Representing the path of the reinforcement in the section.
+            Start of the line defines the first rebar of the configuration, end of the line defines the last rebar.
+
+        Returns
+        -------
+        List[Rebar]
+            List of Rebar objects.
+        """
+        # max(int(n), 1) is used to ensure that at least one rebar is placed
+        rebars = []
+
+        # define the number of rebars based on the length of the line, minimum 1
+        n_rebars = line.length / self.center_to_center
+        n_rebars_applied = max(int(n_rebars), 1)
+
+        # calculate the space between the rebars
+        side_buffer = (line.length - (n_rebars_applied - 1) * self.center_to_center) / 2
+        distances = np.linspace(start=side_buffer, stop=line.length - side_buffer, num=n_rebars_applied)
+
+        # define the representative diameter of the rebar
+        reinforcement_area = 0.25 * np.pi * self.diameter**2 * n_rebars
+        repr_diameter = np.sqrt(reinforcement_area / (0.25 * np.pi * n_rebars_applied))
+
+        for distance in distances:
+            point = line.interpolate(distance)
```

Comment:
Just to check - distance cannot be negative correct? If negative the interpolate will start from the end i believe and might cause unexpected behavior (unless this is the intended usage)

Reply:
correct, but that won't happen because the distance in distances comes from the length of the line itself which is never negative